File: University-Java/src/university/TA.java
package university;

public class TA {
	String name;
	String course;
	String officeHourDay;
	int officeHourTime;
	int recitationTime;

	public TA(String name, String course_name){
		this.name= name;
		this.course=course_name;
		this.officeHourDay="unknown";
		this.officeHourTime = 0;
		//assume all recitations are on Friday. recitation time is an integer.
		this.recitationTime = 0;
	}

	//this makes the assumption that TAs hold office hours in 2hr slots
	public void pickOfficeHours(String day, int start){
		this.officeHourDay=day;
		this.officeHourTime=start;
	}

	public void pickRecitationTime(int start){
		this.recitationTime = start;
	}

	public String getName(){
		return this.name;
	}

	public String getCourse(){
		return this.course;
	}

	public int getRecitationTime(){
		return this.recitationTime;
	}

	public void displayOfficeHours(){
		System.out.println("office hour"+this.officeHourDay+"from"+officeHourTime+"to"+officeHourTime+2);
	}

	//in this model, the TA gives the student their final grade
	public void gradeStudent(Student object_student,String grade){
		object_student.receiveGrade(this.course,grade);	
	}

}

File: University-Python/TA.py
class TA:
    def __init__(self,name,course_name):
        self.name = name
        self.course = course_name
        self.officeHourDay = 'unknown'
        self.officeHourTime = 0
        #assume all recitations are on Friday. recitation time is an integer.
        self.recitationTime = 0

    #this makes the assumption that TAs hold office hours in 2hr slots
    def pickOfficeHours(self, day, start):
        self.officeHourDay = day
        self.officeHourTime = start

    def pickRecitationTime(self, start):
        self.recitationTime = start

    def getName(self):
        return self.name

    def getCourse(self):
        return self.course

    def getRecitationTime(self):
        return self.recitationTime

    def displayOfficeHours(self):
        print('office hours ', self.officeHourDay, 'from', str(self.officeHourTime), 'to', str(self.officeHourTime + 2))


    #in this model, the TA gives the student their final grade
    def gradeStudent(self, student, grade):
        student.receiveGrade(self.course,grade)
